src/modules/iotjs_module_mqtt.h

```diff
@@ -0,0 +1,243 @@
+/* Copyright 2018-present Samsung Electronics Co., Ltd. and other contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef IOTJS_MODULE_MQTT_H
+#define IOTJS_MODULE_MQTT_H
+
+#include "iotjs_def.h"
+
+#define IOTJS_MODULE_MQTT_MAX_REMAINING_LENGTH_BYTES 4
+
+/*
+ * The types of the control packet.
+ * These values determine the aim of the message.
+ */
+enum {
+  CONNECT = 0b0001,
+  CONNACK = 0b0010,
+  PUBLISH = 0b0011,
+  PUBACK = 0b0100,
+  PUBREC = 0b0101,
+  PUBREL = 0b0110,
+  PUBCOMP = 0b0111,
+  SUBSCRIBE = 0b1000,
+  SUBACK = 0b1001,
+  UNSUBSCRIBE = 0b1010,
+  UNSUBACK = 0b1011,
+  PINGREQ = 0b1100,
+  PINGRESP = 0b1101,
+  DISCONNECT = 0b1110
+} iotjs_mqtt_control_packet_type;
+
+/*
+ * The values of the Quality of Service.
+ */
+enum {
+  QoS0 = 0b00, // At most once delivery.
+  QoS1 = 0b01, // At least once delivery.
+  QoS2 = 0b10  // Exactly once delivery.
+} iotjs_mqtt_quality_of_service;
+
+enum {
+  UNACCEPTABLE_PROTOCOL = 1,
+  BAD_IDENTIFIER = 2,
+  SERVER_UNAVIABLE = 3,
+  BAD_CREDENTIALS = 4,
+  UNAUTHORISED = 5
+} iotjs_mqtt_connection_error;
+
+/*
+ * First byte of the message's fixed header.
+ * Contains:
+ * - MQTT Control Packet type,
+ * - Specific flags to each MQTT Control Packet.
+ */
+typedef struct {
+#ifdef MQTT_REVERSED
```

Comment:
How can this be reversed???? There is no bit order, only byte order.

Reply:
This is left over from some old code, we never use it, it's going to get deleted when we clean up the header file.